scipy/linalg/_decomp.py

```diff
@@ -397,6 +397,11 @@ def eigh(a, b=None, *, lower=True, eigvals_only=False, overwrite_a=False,
     often performs worse than the rest except when very few eigenvalues are
     requested for large arrays though there is still no performance guarantee.
 
+    Note that the underlying LAPACK algorithms are different depending on whether
+    `eigvals_only` is True or False --- thus the eigenvalues may differ
+    depending on whether eigenvectors are requested or not. The difference is
```

Comment:
The difference is in general of the order of the machine epsilon times the largest singular value of the matrix.

Reply:
Thank you @christoph-conrads ! Updated to the suggestion.